wsmaster/codenvy-hosted-platform-api-impl/src/test/java/com/codenvy/api/workspace/LimitsCheckingWorkspaceManagerTest.java

```diff
@@ -454,11 +471,104 @@ public void shouldPermitToCheckRamOnlyForFiveThreadsAtTheSameTime() throws Excep
         executor.submit(runnable);
 
         //Wait for throughput limit will be reached and check that RAM check was performed only in allowed number of threads.
-        verify(manager, timeout(300).times(5)).checkSystemRamAndPropagateStart(anyObject());
+        verify(manager, timeout(300).times(5)).checkLimitsAndPropagateStart(anyString(), anyObject());
 
         //Execute paused threads to release the throughput limit for other threads.
         invokeProcessLatch.countDown();
         //Wait for throughput limit will be released and check that RAM check was performed in other threads.
-        verify(manager, timeout(300).times(7)).checkSystemRamAndPropagateStart(anyObject());
+        verify(manager, timeout(300).times(7)).checkLimitsAndPropagateStart(anyString(), anyObject());
+    }
+
+    @Test
+    public void shouldCallCreateCallBackIfStartedWorkspacesNumberLimitIsNotExceeded() throws Exception {
+        final LimitsCheckingWorkspaceManager manager = spy(new LimitsCheckingWorkspaceManager(2,
```

Comment:
There are so much parameters, this tests class grows really fast.
What do  you think about creating a builder class for testing like `ManagerBuilder` with a method like `createDefault()` so it would be possible to change only parameters which are tested, eg:

```java
ManagerBuilder.createDefault()
              .setRunningLimit(2)
              .build();
```

This may simplify tests a bit.